stl/inc/mdspan

```diff
@@ -786,7 +786,7 @@ private:
     template <class... _IndexTypes, size_t... _Seq>
     _NODISCARD constexpr index_type _Index_impl(index_sequence<_Seq...>, _IndexTypes... _Indices) const noexcept {
         _STL_INTERNAL_STATIC_ASSERT((same_as<_IndexTypes, index_type> && ...));
-        return ((_Indices * _Strides[_Seq]) + ... + 0);
+        return static_cast<index_type>(((_Indices * _Strides[_Seq]) + ... + 0));

```

Comment:
Is the issue here just the (implicitly `int`) zero in the fold init? `_Strides` is an `array<index_type>` and `_Indices` is casted to `index_type` at the call site. If that's the case, then maybe
```suggestion
        return ((_Indices * _Strides[_Seq]) + ... + index_type{0});
```
would avoid any integer promotions.

Reply:
Never mind, I understand now the integer promotions happen anyway, even when the types match.